src/Admin/AbstractAdmin.php

```diff
@@ -3340,6 +3337,26 @@ private function buildRoutes(): void
             $extension->configureRoutes($this, $this->routes);
         }
     }
+
+    private function createFieldDescription(AdminInterface $admin, string $name, array $options = []): FieldDescriptionInterface
+    {
+        $fieldDescriptionFactory = $this->getFieldDescriptionFactory();
+
+        // NEXT_MAJOR: Remove the "if" block and leave the "else" one.
+        if (null === $fieldDescriptionFactory) {
+            $fieldDescription = $this->getModelManager()->getNewFieldDescriptionInstance(
+                $admin->getClass(),
+                $name,
+                $options
+            );
+        } else {
+            $fieldDescription = $this->fieldDescriptionFactory->create($admin, $name, $options);
+        }
+
+        $fieldDescription->setAdmin($this);
```

Comment:
This should be in the `if` part.
It's the fieldDescriptionFactory job to set the Admin, no ?